compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/ResolveUtils.kt

```diff
@@ -349,3 +352,46 @@ fun BodyResolveComponents.initialTypeOfCandidate(candidate: Candidate): ConeKotl
 private fun initialTypeOfCandidate(candidate: Candidate, typeRef: FirResolvedTypeRef): ConeKotlinType {
     return candidate.substitutor.substituteOrSelf(typeRef.type)
 }
+
+inline val FirCallableDeclaration<*>.containingClass: FirRegularClass?
+    get() = this.containingClassAttr?.let { lookupTag ->
+        session.symbolProvider.getSymbolByLookupTag(lookupTag)?.fir as? FirRegularClass
+    }
+
+val FirFunction<*>.asBadForNamedArgumentTarget: BadNamedArgumentsTarget?
+    get() {
+        if (this is FirConstructor && this.isPrimary) {
+            this.containingClass?.let { containingClass ->
+                if (containingClass.classKind == ClassKind.ANNOTATION_CLASS) {
+                    // Java annotation classes allow (actually require) named parameters.
+                    return null
+                }
+            }
+        }
+        return when (origin) {
+            FirDeclarationOrigin.Source, FirDeclarationOrigin.Library, FirDeclarationOrigin.BuiltIns -> null
+            FirDeclarationOrigin.Delegated -> delegatedWrapperData?.wrapped?.asBadForNamedArgumentTarget
+            FirDeclarationOrigin.ImportedFromObject -> importedFromObjectData?.original?.asBadForNamedArgumentTarget
+            // FE1.0 actually only cares about whether the first interface participating the intersection has stable parameter or not. So
+            // we just do the same rather than trying to figure out all the interfaces participating the intersection and decide based on
+            // that.
```

Comment:
I'm afraid it's not exactly same.

https://github.com/JetBrains/kotlin/blob/master/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/scopes/impl/FirTypeIntersectionScope.kt#L286
https://github.com/JetBrains/kotlin/blob/master/compiler/fir/resolve/src/org/jetbrains/kotlin/fir/scopes/impl/FirTypeIntersectionScope.kt#L305

It points to the most specific one, not the _first_ one. So, where is the counter part in FE 1.0 that looks up "first" interface?

Reply:
I was being naive and inferred the behavior through my observations. So it looks like the corresponding FE1.0 logic is in org.jetbrains.kotlin.resolve.OverridingUtil#selectMostSpecificMember, and it appears to be the same logic. So I should rephrase my comment here mention this.